### PR TITLE
Saas boost toolkit

### PR DESCRIPTION
**Pull request by LYH-RAIN**
feat: Introduce commodity management and refine currency handling

This pull request includes changes for two main enhancements:

1. Support for Commodity and Specification Management:
   - Enable the creation of products along with package options, and manage their lifecycle.
   - Integrate with Compute Nest services to bind products with their respective services.
   - Introduce functionality to activate and deactivate products for public listing.

2. Refactor Currency Representation:
   - Transition all monetary values in the codebase to be represented using BigDecimal, ensuring precision.
   - Modify the underlying storage mechanism for currency-related fields, moving from double to long type. 
     This allows for storage of monetary amounts in minor units (cents).
   - Update front-end display logic to convert long-type monetary values from cents to yuan for user presentation.

These combined changes create a cohesive feature set for managing commodities and offer accurate currency handling across the platform.

### CODE REVIEW

File: boost.common/src/main/java/org/example/common/constant/PayPeriodUnit.java [hunk 1]
```diff
@@ -20,24 +20,33 @@ public enum PayPeriodUnit {
     /**
      * Monthly subscription.
      */
-    Month("月"),
+    Month("month"),
```
Comment (Alicloud-ComputeNest) [inline boost.common/src/main/java/org/example/common/constant/PayPeriodUnit.java]: month要大写吗

File: boost.common/src/main/java/org/example/common/dataobject/OrderDO.java [hunk 2]
```diff
@@ -72,22 +75,27 @@ public class OrderDO {
     /**
      * Buyer payment amount.
      */
-    private Double buyerPayAmount;
+    private Long buyerPayAmount;
```
Comment (Alicloud-ComputeNest) [inline boost.common/src/main/java/org/example/common/dataobject/OrderDO.java:78]: 咋改成long了

File: boost.common/src/main/java/org/example/common/config/WebConfig.java [hunk 3]
```diff
@@ -48,7 +48,10 @@ public void addInterceptors(InterceptorRegistry registry) {
                 .addPathPatterns("/api/**");
         registry.addInterceptor(tokenAuthenticationInterceptor)
                 .addPathPatterns("/api/**")
-                .excludePathPatterns("/api/getAuthToken").excludePathPatterns("/api/getAuthConfiguration");
```
Comment (Alicloud-ComputeNest) [inline boost.common/src/main/java/org/example/common/config/WebConfig.java:51]: 服务商的接口是怎么鉴权的，也是token吗，然后判断是服务商aliUid?

File: boost.common/src/main/java/org/example/common/dataobject/CommoditySpecificationDO.java [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+
+    /**
+     * Unit for pricing calculation (e.g., Year, Month, or PostPaid).
+     */
+    private PayPeriodUnit payPeriodUnit;
+
+    /**
+     * Allowed payment durations for subscription-based scenarios, such as [1, 2, 3] for years or months.
+     */
+    private String payPeriods;
```
Comment (Alicloud-ComputeNest) [inline boost.common/src/main/java/org/example/common/dataobject/CommoditySpecificationDO.java:57]: 这个字段为啥是String 看着是个数组？

Comment (Alicloud-ComputeNest) [reply boost.common/src/main/java/org/example/common/dataobject/CommoditySpecificationDO.java:57]: List<Long> ?

File: boost.common/src/main/java/org/example/common/helper/SpiTokenHelper.java [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+    private AcsApiCaller acsApiCaller;
+
+    @Resource
+    private CommodityOtsHelper commodityOtsHelper;
+
+    private static final String GET_SERVICE_PROVIDER_KEY = "GetServiceProviderKey";
+
+    private static final String SERVICE_KEY = "ServiceKey";
+
+    public void checkSpiToken(Object param, String spiToken, String commodityCode) {
```
Comment (Alicloud-ComputeNest) [inline boost.common/src/main/java/org/example/common/helper/SpiTokenHelper.java]: 没看到这个接口哪里调用的呢，是不是漏了

File: boost.common/src/main/java/org/example/common/dataobject/OrderDO.java (hunk 2 above)
Comment (LYH-RAIN) [reply boost.common/src/main/java/org/example/common/dataobject/OrderDO.java:78]: 这块儿最近做了个调研。java里关于金钱的都不能用double，因为会存在精度丢失的问题，比如double的0.01实际是0.01xxxxxx21。所以严格来说钱涉及计算都采用BigDecimal,钱存储会转换为分存储，比如凌霄就是这样。

File: boost.common/src/main/java/org/example/common/config/WebConfig.java (hunk 3 above)
Comment (LYH-RAIN) [reply boost.common/src/main/java/org/example/common/config/WebConfig.java:51]: 通过token。目前是把管理员aliuid写进配置文件里，对于加了@admin注解的接口，通过注入的拦截器判断是否是管理员访问。这块儿后续也可以把管理员aliuid放参数管理里。

File: boost.common/src/main/java/org/example/common/dataobject/CommoditySpecificationDO.java (hunk 4 above)
Comment (LYH-RAIN) [reply boost.common/src/main/java/org/example/common/dataobject/CommoditySpecificationDO.java:57]: 作为字符串存储是因为ots对list的支持不太友好，所以我直接放的String。

File: boost.common/src/main/java/org/example/common/helper/SpiTokenHelper.java (hunk 5 above)
Comment (LYH-RAIN) [reply boost.common/src/main/java/org/example/common/helper/SpiTokenHelper.java]: 在orderService的createOrder和commodityService的getCommodity和getCommodityPrice接口都有调用。

File: boost.common/src/main/java/org/example/common/constant/ChargeType.java [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+ *See the License for the specific language governing permissions and
+ *limitations under the License.
+ */
+
+package org.example.common.constant;
+
+import lombok.Getter;
+
+@Getter
+public enum ChargeType {
```
Comment (linziqiaolzq) [inline boost.common/src/main/java/org/example/common/constant/ChargeType.java:21]: 其实这里不用枚举，就2个弄个常量就行，如果不好改这样也行

File: boost.common/src/main/java/org/example/common/constant/CommoditySpecificationOtsConstant.java [hunk 7]
```diff
@@ -0,0 +1,73 @@
...
+    /**
+     * Field for commodity status on CommoditySpecification table.
+     */
+    String COMMODITY_STATUS = "commodityStatus";
+
+    /**
+     * Set of fields commonly used for range queries in the search index for commodity specifications.
+     * Assuming we have fields that represent time or other quantities that may require range queries.
+     */
+    Set<String> QUERY_FILTERS_SET = Collections.unmodifiableSet(new HashSet<String>() {
```
Comment (linziqiaolzq) [inline boost.common/src/main/java/org/example/common/constant/CommoditySpecificationOtsConstant.java]: 一行就行～

File: boost.common/src/main/java/org/example/common/constant/OrderType.java [hunk 8]
```diff
@@ -0,0 +1,38 @@
...
+ *WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *See the License for the specific language governing permissions and
+ *limitations under the License.
+ */
+
+package org.example.common.constant;
+
+public enum OrderType {
+
+    BUY("Purchase an item"),
```
Comment (linziqiaolzq) [inline boost.common/src/main/java/org/example/common/constant/OrderType.java]: 就好奇，这个OrderType咋会是这样

File: boost.common/src/main/java/org/example/common/helper/SpiTokenHelper.java [hunk 9]
```diff
@@ -0,0 +1,83 @@
...
+                String serviceKey = map.getOrDefault(SERVICE_KEY, null);
+                String currentSpiToken = TokenUtil.createSpiToken(param, serviceKey);
+                if (StringUtils.isEmpty(currentSpiToken) || !currentSpiToken.equals(spiToken)) {
+                    throw new BizException(ErrorInfo.SPI_TOKEN_VALIDATION_FAILED.getStatusCode(), ErrorInfo.SPI_TOKEN_VALIDATION_FAILED.getCode(), String.format(ErrorInfo.SPI_TOKEN_VALIDATION_FAILED.getMessage(), spiToken));
+                }
+                return;
+            }
+            throw new BizException(ErrorInfo.SERVICE_PROVIDER_KEY_NOT_EXIST.getStatusCode(), ErrorInfo.SERVICE_PROVIDER_KEY_NOT_EXIST.getCode(), String.format(ErrorInfo.SERVICE_PROVIDER_KEY_NOT_EXIST.getMessage(), commodity.getServiceId(), commodityCode));
+        } catch (ClientException e) {
+            throw new BizException(ErrorInfo.COMMON_REQUEST_FAILED.getStatusCode(), e.getErrCode(), String.format(ErrorInfo.COMMON_REQUEST_FAILED.getMessage(), GET_SERVICE_PROVIDER_KEY, JsonUtil.toJsonString(commonRequest), e.getMessage()), e);
```
Comment (linziqiaolzq) [inline boost.common/src/main/java/org/example/common/helper/SpiTokenHelper.java]: 换个行，这行太长了

File: boost.common/src/main/java/org/example/common/model/CommodityPriceModel.java [hunk 10]
```diff
@@ -12,34 +12,46 @@
...
+@Data
+public class CommodityPriceModel {
 
-public class PayOrderModel {
+    /**
+     * 单价
+     */
+    private Long unitPrice;
+
+    /**
```
Comment (linziqiaolzq) [inline boost.common/src/main/java/org/example/common/model/CommodityPriceModel.java:28]: 我记得你前面好像注解是英文吧

Comment (LYH-RAIN) [reply boost.common/src/main/java/org/example/common/model/CommodityPriceModel.java:28]: 都换为英文了

**Pull request merged**